docs/src/main/sphinx/installation/query-resiliency.rst

```diff
@@ -0,0 +1,119 @@
+causing the query to fail.
+
+When configured, the Trino cluster buffers data used by workers
+during query processing. If a worker node fails for any reason, such as a
+network outage or running out of available resources, the coordinator issues the
+data to an available worker that can pick up query processing, allowing the
+query to continue.
+
+Architecture
+------------
+
+The coordinator node uses a configured exchange manager service that buffers
+data for query processing in an external location such as an S3 bucket. Worker
+nodes send data to the buffer as they execute their query tasks.
+
+Best practices and considerations
+---------------------------------
+
+A fault-tolerant cluster is best suited for large batch queries. Fault-tolerant
+execution of a high volume of short-running may experience increased latency. As
+such, it is recommended to run a dedicated fault-tolerant Trino
+cluster for handling batch operations, separate from a cluster that is
+designated for a higher query volume.
+
+Catalogs for the following data sources support fault-tolerant execution of read
+and write operations:
+
+* Delta Lake
+* Hive
+* Iceberg
+* MySQL
+* PostgreSQL
+* SQL Server
+
+Catalogs for other data sources only support fault-tolerant execution of read
+operations. When fault-tolerant execution is enabled on a cluster, write
+operations are disabled for any catalogs that do not support fault-tolerant
+execution of those operations.
+
+The exchange manager may send a large amount of data to the exchange storage,
+resulting in high I/O load on that storage. You can configure multiple storage
+locations for use by the exchange manager to help balance the I/O load between
+them.
+
+Configuration
+-------------
+
+The following steps describe how to configure a Trino cluster for
+fault-tolerant execution with an S3 bucket-based exchange:
+
+1. Set up an S3 bucket to use as the exchange storage. For this example we are
+   using an AWS S3 bucket, but other storage options are described in the
+   :doc:`reference documentation </admin/fault-tolerant-execution>`
+   as well. You can use multiple S3 buckets for exchange storage.
+
+   For each bucket in AWS, collect the following information:
+
+   * S3 URI location for the bucket, such as ``s3://exchange-spooling-bucket``
+
+   * Region that the bucket is located in, such as ``us-west-1``
+
+   * AWS access and secret keys for the bucket
+
+2. For a :doc:`Kubernetes deployment of Trino </installation/kubernetes>`, add
+   the following exchange manager configuration in the
+   ``server.exchangeManager`` and ``additionalExchangeManagerProperties``
+   sections of the Helm chart, using the gathered S3 bucket information:
+
+   .. code-block:: yaml
+
+       server:
+         exchangeManager:
+           name=filesystem
+           base-directories=s3://exchange-spooling-bucket-1,s3://exchange-spooling-bucket-2
+
+       additionalExchangeManagerProperties:
+         s3.region=us-west-1
```

Comment:
So the same credentials and region need to apply to all S3 paths.. ?

Reply:
I believe so, @arhimondr can you confirm?